### PR TITLE
android: Advanced server config

### PR DESCRIPTION
**Pull request by avently**

### CODE REVIEW

File: apps/android/app/src/main/java/chat/simplex/app/views/usersettings/ScanSMPServer.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+@Composable
+private fun ScanSMPServerLayout(onNext: (ServerCfg) -> Unit) {
+  Column(
+    Modifier
+      .fillMaxSize()
+  ) {
+    AppBarTitle(stringResource(R.string.smp_servers_scan_qr))
+    QRCodeScanner { text ->
```
Comment (jr-simplex) [inline apps/android/app/src/main/java/chat/simplex/app/views/usersettings/ScanSMPServer.kt:25]: seems like it doesn't ask/check for camera permission and scanner stays black

File: apps/android/app/src/main/java/chat/simplex/app/views/usersettings/ScanSMPServer.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        onNext(ServerCfg(text, false, null, true))
+      } else {
+        AlertManager.shared.showAlertMsg(
+          title = generalGetString(R.string.smp_servers_invalid_address),
+          text = generalGetString(R.string.smp_servers_check_address)
+        )
+      }
+    }
+  }
+}
```
Comment (jr-simplex) [inline apps/android/app/src/main/java/chat/simplex/app/views/usersettings/ScanSMPServer.kt:37]:
```suggestion
}

```

File: apps/android/app/src/main/java/chat/simplex/app/views/usersettings/SMPServerView.kt [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      .fillMaxWidth()
+      .verticalScroll(rememberScrollState())
+      .padding(bottom = DEFAULT_PADDING)
+  ) {
+    AppBarTitle(stringResource(R.string.your_SMP_servers))
+
+    if (server.preset) {
+      PresetServer(testing, server, testServer, onUpdate, onDelete)
+    } else {
+      CustomServer(testing, server, testServer, onUpdate, onDelete)
```
Comment (jr-simplex) [inline apps/android/app/src/main/java/chat/simplex/app/views/usersettings/SMPServerView.kt:86]: preset and custom servers have different font

File: apps/android/app/src/main/java/chat/simplex/app/views/usersettings/SMPServerView.kt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  val serverAddress = remember { mutableStateOf(server.server) }
+  val valid = remember { derivedStateOf { parseServerAddress(serverAddress.value)?.valid == true } }
+  SectionView(
+    stringResource(R.string.smp_servers_your_server_address),
+    icon = Icons.Outlined.Block,
+    iconTint = if (!valid.value) MaterialTheme.colors.error else Color.Transparent,
+  ) {
+    TextEditor(Modifier.height(144.dp), text = serverAddress, border = false) {
+      onUpdate(server.copy(server = serverAddress.value))
+    }
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply apps/android/app/src/main/java/chat/simplex/app/views/usersettings/SMPServerView.kt]: yes

**Pull request merged**